### PR TITLE
[UI] Implement Add Expense UI

### PR DESCRIPTION
**Pull request by louisepujante**
## Related Links:
- [Issue link](https://www.notion.so/Transaction-Implement-Add-Expense-UI-c23c0c6119f94f0685411e8eeaf24320)

## Definition of Done
- [x]  Implement expense form with following input fields (Name, Amount, Category Date)
- [x]  Add button below all the input fields
- [x]  Use reusable component for all the elements inside the form
- [x]  Add transaction page appears when floating action button is clicked (Thank you for helping, Ervin 🙇‍♀️ )

## Notes:
- form is scrollable
- onSubmit can be changed

## Screenshots
image
image

## Test View  Points
- [x] add transaction page should appear when floating action button is clicked
- [x] all fields must be provided
     - [x] show error message for empty fields
- [x] form must show in "Add expense" tab
- [x] values in dialog must match user input

### CODE REVIEW

File: lib/views/widgets/input/date_field.dart [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-    required this.lastDate,
-  }) : super(key: key);
+  final TextEditingController dateController;
+
+  DateField(
+      {Key? key,
+      required this.firstDate,
+      required this.lastDate,
+      required this.dateController})
+      : super(key: key);
```
Comment (inahsanjuan) [inline lib/views/widgets/input/date_field.dart]: Fix indentation same as before
https://github.com/framgia/sph_flutter-capstone/pull/19/files#diff-b7cb3161fa0f9324d32e0ae025677cccc6c00ac757578a82dfa660f395809013L8-L12

File: lib/views/pages/transactions/add_expense.dart [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                input: InputField(
+                  hintText: '150.00',
+                  inputType: TextInputType.number,
+                  inputController: formInputControllers['amountController']!,
+                ),
+              ),
+              InputGroup(
+                  label: 'CATEGORY',
+                  input: SelectInput(
+                    items: const ['Food and Drink', 'Bills', 'Other'], // TODO: Change items here
```
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]: Move to a variable, to easily update later

File: lib/views/widgets/navigation/floating_action_button.dart [hunk 3]
```diff
@@ -1,16 +1,27 @@
 import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:sun_flutter_capstone/consts/global_style.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:sun_flutter_capstone/utils/routes/router.gr.dart';
 
-class CenterActionButton extends StatelessWidget {
+class CenterActionButton extends HookConsumerWidget {
```
Comment (inahsanjuan) [inline lib/views/widgets/navigation/floating_action_button.dart:7]: @louisepujante Asking so I can also learn 😄 
What is this change for?

File: lib/views/widgets/tabs/tab_buttons.dart [hunk 4]
```diff
@@ -23,29 +21,26 @@ class TabButtons extends StatelessWidget {
...
+      child: Padding(
+        padding: const EdgeInsets.all(5),
+        child: TabBar(
+          controller: tabController,
+          indicator: BoxDecoration(
               borderRadius: BorderRadius.circular(
                 25.0,
               ),
-              color: index == 0 ? AppColor.pink : AppColor.darkBlue,
+              color: index == 0 ? AppColor.pink : AppColor.darkBlue),
```
Comment (inahsanjuan) [inline lib/views/widgets/tabs/tab_buttons.dart]: to trigger formatting
```suggestion
              color: index == 0 ? AppColor.pink : AppColor.darkBlue,),
```

File: lib/views/pages/transactions/add_expense.dart [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                    initialValue: 'Food and Drink',
+                    valueReceiver: (String value) {
+                      formInputControllers['categoryController']!.text = value;
+                    },
+                  )),
```
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]: trigger formatting
```suggestion
                  ),),
```

File: lib/views/pages/transactions/add_expense.dart [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                label: 'NAME',
+                input: InputField(
+                  hintText: 'Starbucks',
+                  inputController: formInputControllers['nameController']!,
+                ),
+              ),
+              InputGroup(
+                label: 'AMOUNT',
+                input: InputField(
+                  hintText: '150.00',
```
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]: Remove all hint text in this Form, UI shows sample value not hint text.

File: lib/views/pages/transactions/add_expense.dart [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      padding: const EdgeInsets.symmetric(horizontal: 25, vertical: 40),
+      content: Form(
+        key: expenseFormKey,
+        child: SingleChildScrollView(
+          child: Column(
+            children: [
+              InputGroup(
+                label: 'NAME',
+                input: InputField(
+                  hintText: 'Starbucks',
```
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]:
```suggestion
                  hintText: 'What are you spending for?',
```

File: lib/views/pages/transactions/add_expense.dart (hunk 6 above)
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]:
```suggestion
                  hintText: 'How much did you spend?',
```

File: lib/views/pages/transactions/add_expense.dart [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+                  hintText: '150.00',
+                  inputType: TextInputType.number,
+                  inputController: formInputControllers['amountController']!,
+                ),
+              ),
+              InputGroup(
+                  label: 'CATEGORY',
+                  input: SelectInput(
+                    items: const ['Food and Drink', 'Bills', 'Other'], // TODO: Change items here
+                    initialValue: 'Food and Drink',
```
Comment (inahsanjuan) [inline lib/views/pages/transactions/add_expense.dart]: Initial value should be empty

**Review by inahsanjuan — CHANGES_REQUESTED** [review]
Other behavior that needs fixing, but we can create new task for this just noting it here
* Add button in navbar can be clicked multiple times, and creates an open form multiple times as well
* When field is on focus, text is not blue

File: lib/views/widgets/input/input_field.dart [hunk 9]
```diff
@@ -1,16 +1,19 @@
 import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
 
 class InputField extends StatelessWidget {
   final String hintText;
-  final bool isNumber;
   final String errorMessage;
+  final TextInputType inputType;
+  final bool isPassword;
```
Comment (rdelrosariojr) [inline lib/views/widgets/input/input_field.dart]: @louisepujante I think no need na for isPassword, we don't have password field to our design.

File: lib/views/widgets/navigation/floating_action_button.dart (hunk 3 above)
Comment (louisepujante) [reply lib/views/widgets/navigation/floating_action_button.dart:7]: This was added by Ervin, I think it's to trigger the routing.. same with the bottom nav bar buttons which also uses this widget

**Comment by louisepujante** [comment]
Changes requested:

- [x]  fix indentation of date field
- [x]  move dropdown items value to a variable
    - [x]  initial value of dropdown should be empty
    - [x]  validation if no items were selected
- [x]  formatting in tab buttons
- [x]  formatting in add expense
- [x]  change hint text
- [x]  change text color on focus of input

**Review by inahsanjuan — APPROVED** [review]
Review time: 20 min
Comments: 8

**Pull request merged**